deps/v8/src/compiler/turboshaft/machine-optimization-reducer.h

```diff
@@ -1349,53 +1349,23 @@ class MachineOptimizationReducer : public Next {
         if (matcher.MatchConstantShiftRightArithmeticShiftOutZeros(
                 left, &x, rep_w, &k1) &&
             matcher.MatchIntegralWordConstant(right, rep_w, &k2) &&
-            CountLeadingSignBits(k2, rep_w) > k1) {
-          if (matcher.Get(left).saturated_use_count.IsZero()) {
-            return __ Comparison(
-                x, __ WordConstant(base::bits::Unsigned(k2) << k1, rep_w), kind,
-                rep_w);
-          } else if constexpr (reducer_list_contains<
-                                   ReducerList, ValueNumberingReducer>::value) {
-            // If the shift has uses, we only apply the transformation if the
-            // result would be GVNed away.
-            OpIndex rhs =
-                __ WordConstant(base::bits::Unsigned(k2) << k1, rep_w);
-            static_assert(ComparisonOp::input_count == 2);
-            static_assert(sizeof(ComparisonOp) == 8);
-            base::SmallVector<OperationStorageSlot, 32> storage;
-            ComparisonOp* cmp =
-                CreateOperation<ComparisonOp>(storage, x, rhs, kind, rep_w);
-            if (__ WillGVNOp(*cmp)) {
-              return __ Comparison(x, rhs, kind, rep_w);
-            }
-          }
+            CountLeadingSignBits(k2, rep_w) > k1 &&
+            matcher.Get(left).saturated_use_count.IsZero()) {
+          return __ Comparison(
+              x, __ WordConstant(base::bits::Unsigned(k2) << k1, rep_w), kind,
+              rep_w);
```

Comment:
This change seems non-trivial. What's the effect of removing the `else if` branch?

Reply:
That is a revert of [this commit](https://github.com/v8/v8/commit/8005bd282bf1565a11cae24a9811e99f8b3fc64b#diff-f11fdde01fe8fbd8c3fa045a3aadf1d6b06d0d442a6e6336626cd0b4361cced0). It was also a part of [v12.2 fix](https://github.com/nodejs/node/commit/b9d806a2dd7d6b137cd9f0099ec0f8ed0abea4d6#diff-ba5f45f27bb33b78f2da3ec5ade73dd18f61046810dbce4cca7ecad133fe7504) before.